### PR TITLE
Simplify tf.keras.activations.softmax codepath

### LINKED ISSUE #35199: Softmax activations don't get converted to Softmax TFLite operator if ndim > 2
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 10.15.1
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source): tf-nightly==2.1.0.dev20191203

**Command used to run the converter or code if you’re using the Python API**

```
import pathlib

inpt = tf.keras.layers.Input(shape=[256, 256, 3])
out = tf.keras.layers.Lambda(lambda x: tf.keras.activations.softmax(x))(inpt)
out = tf.keras.layers.Lambda(lambda x: tf.nn.softmax(x))(out)
model = tf.keras.Model(inpt, out)

converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()
pathlib.Path('out.tflite').write_bytes(tflite_model)
```

**Failure details**
image
This graph shows the difference between the different softmax methods.  When using `tf.keras.activations.softmax`, there is [code](https://github.com/tensorflow/tensorflow/blob/v2.1.0-rc1/tensorflow/python/keras/activations.py#L43-L79) with a workaround for multiple dimensions.  It looks like this was written before the tensorflow op had multi-dimension support.

### PR DESCRIPTION
**Pull request by WindQAQ**
Fixes https://github.com/tensorflow/tensorflow/issues/35199 although TFLite has already optimized that pattern into single softmax op. Since `tf.nn.softmax` supports input of any rank, there is no reason to have other implementations. For common 3D input with `axis=-1`, `tf.nn.softmax` is also faster than original implementation.

https://colab.research.google.com/drive/1E1rS69LJ7e-pH4bcyfxEJ_wC6IiRISvS?usp=sharing

### CODE REVIEW

**Review by vodiylik — APPROVED** [review]

File: tensorflow/python/keras/activations_test.py [hunk 1]
```diff
@@ -31,10 +31,10 @@
 from tensorflow.python.platform import test
 
 
-def _ref_softmax(values):
-  m = np.max(values)
+def _ref_softmax(values, axis=-1):
```
Comment (qlzh727) [inline tensorflow/python/keras/activations_test.py:34]: I think you can skip axis since its never changed by the caller.

**Review by qlzh727 — APPROVED** [review]
Thanks for the fix and sorry for the late reply. Just a nit about the test.

File: tensorflow/python/keras/activations_test.py (hunk 1 above)
Comment (WindQAQ) [reply tensorflow/python/keras/activations_test.py:34]: Sure let me modify it real quick.

Comment (WindQAQ) [reply tensorflow/python/keras/activations_test.py]: Done 😃

**Review by qlzh727 — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by qlzh727** [comment]
Hum, it seems that this PR gets rolled back since it breaks one of the internal test. Let me take a look and fix forward.
